frontend/react/src/components/layout/DateRange.js

```diff
@@ -1,296 +1,332 @@
+  if (input.length > 2) {
+    returnString = "Month length must not exceed 2";
+  }
+
+  // Checks for non-numeric characters
+  if (Number.isNaN(parseInt(input, 10)) || /^\d+$/.test(input) === false) {
+    returnString = "Please enter a number";
+  }
+
+  if (parseInt(input, 10) < 1 || parseInt(input, 10) > 12) {
+    // Checks that the month value is within a normal range
+    returnString = "Please enter a valid month number";
+  }
+  return returnString;
+};
+
+class DateRange extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      endRangeErr: false,
+      monthStart: "",
+      yearStart: "",
+      monthEnd: "",
+      yearEnd: "",
+      startErrorMessage: [],
+      endErrorMessage: [],
+    };
+    this.handleInput = this.handleInput.bind(this);
+
+    this.checkChronology = this.checkChronology.bind(this);
+
+    this.validateStartInput = this.validateStartInput.bind(this);
+    this.validateEndInput = this.validateEndInput.bind(this);
+
+    this.validateYear = this.validateYear.bind(this);
+  }
+
+  componentDidMount() {
+    // Stored value example: ['2019-11-01', '2020-09-01']
+    const { question } = this.props;
+    const storedValue = question.answer.entry;
+
+    if (storedValue) {
+      // Returns an array with the values of the start date, split
+      const startDate = storedValue[0].split("-"); // ie: ['2019', '11', '01']
+
+      // Returns an array with the values of the start date, split
+      const endDate = storedValue[1].split("-"); // ie: ['2020', '09', '01']
+
+      const monthStartValue = startDate[1]; // '11'
+      const yearStartValue = startDate[0]; // '2019'
+
+      const monthEndValue = endDate[1]; // '09'
+      const yearEndValue = endDate[0]; // '2020'
+
+      this.setState({
+        monthStart: monthStartValue || "",
+        monthEnd: monthEndValue || "",
+        yearStart: yearStartValue || "",
+        yearEnd: yearEndValue || "",
+      });
+    } else {
+      this.setState({
+        monthStart: "",
+        monthEnd: "",
+        yearStart: "",
+        yearEnd: "",
+      });
+    }
+  }
+
+  // This method checks all 4 fields to confirm that the start range is before the end range
+  checkChronology() {
+    const { onChange, question } = this.props;
+    const {
+      yearStart,
+      yearEnd,
+      startErrorMessage,
+      endErrorMessage,
+    } = this.state;
+    const errorCheck = [...startErrorMessage, ...endErrorMessage]; // Array of all input errors in state
+
+    let { monthStart, monthEnd } = this.state;
+    let chronologyError;
+
+    // Ensure that all 4 fields are filled in
+    if (monthStart && monthEnd && yearStart && yearEnd) {
+      // Turn the input into date objects for easy comparison
+
+      monthStart = monthStart.padStart(2, "0");
+      monthEnd = monthEnd.padStart(2, "0");
+
+      const startDate = new Date(yearStart, monthStart - 1);
+      const endDate = new Date(yearEnd, monthEnd - 1);
```

Comment:
I'd suggest moving this to after you create the `Date` objects. I'm like 98% sure it's fine as you have it, but it feels weird to me to pad the number with zeros and then do math on it. It ***feels*** like a bug (even though I'm pretty sure it's not), which might cause a future version of you to slow down when looking at this code.

```suggestion
      const startDate = new Date(yearStart, monthStart - 1);
      const endDate = new Date(yearEnd, monthEnd - 1);

      monthStart = monthStart.padStart(2, "0");
      monthEnd = monthEnd.padStart(2, "0");
```

Reply:
Makes sense, taken care of!